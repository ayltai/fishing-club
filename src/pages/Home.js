// @flow
'use strict';

import React from 'react';
import muiThemeable from 'material-ui/styles/muiThemeable';
import { Card, CardMedia, CardTitle, CardText } from 'material-ui/Card';
import LogoImage from '../images/logo.jpg';

class Home extends React.Component {
    componentDidMount() : void {
        window.scrollTo(0, 0);
    }

    render() : any {
        return (
            <div>
                <Card className="homeItem">
                    <CardMedia>
                        <img
                            src={LogoImage}
                            style={{
                                maxWidth : '100%'
                            }} />
                    </CardMedia>
                    <CardTitle title="Fishing Club" />
                    <CardText>
                        <p>Welcome to our fishing club! We are going to hold our first ever fishing activity. Here you will find useful information in a nicely organized way.</p>
                        <p>We try to include as much detailed information as we can. Do let us know if we miss any information that you need. If you are in a hurry, please check <a href="/others/preparations" title="Preparations">Preparations</a> page where you will learn how to better prepare yourself for the activity.</p>
                        <p>Hope you will enjoy the activity.</p>
                        <br />
                        <p><small>The source code of this website is hosted on <a href="https://github.com/ayltai/fishing-club" title="GitHub">GitHub</a>.</small></p>
                    </CardText>
                </Card>
            </div>
        );
    }
}

export default muiThemeable()(Home);
