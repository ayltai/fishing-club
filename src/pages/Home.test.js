'use strict';

import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import Home from './Home';

it('renders correctly', () => {
    const div = document.createElement('div');
    ReactDOM.render(
        <BrowserRouter>
            <MuiThemeProvider>
                <Home />
            </MuiThemeProvider>
        </BrowserRouter>,
        div);

    expect(div.innerHTML).toMatchSnapshot();
});
