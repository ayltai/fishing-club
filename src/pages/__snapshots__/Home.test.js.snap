// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `

<div data-reactroot>
  <div class="homeItem"
       style="color: rgba(0, 0, 0, 0.87); background-color: rgb(255, 255, 255); box-sizing: border-box; font-family: Roboto, sans-serif; box-shadow: 0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12); border-radius: 2px; z-index: 1;"
  >
    <div style="padding-bottom: 8px;">
      <div style="position: relative;">
        <div>
          <img src="logo.jpg"
               style="vertical-align: top; max-width: 100%; min-width: 100%; width: 100%;"
          >
        </div>
        <!-- react-text: 7 -->
        <!-- /react-text -->
      </div>
      <div style="padding: 16px; position: relative;">
        <span style="font-size: 24px; color: rgba(0, 0, 0, 0.87); display: block; line-height: 36px;">
          Fishing Club
        </span>
        <span style="font-size: 14px; color: rgba(0, 0, 0, 0.54); display: block;">
        </span>
      </div>
      <div style="padding: 16px; font-size: 14px; color: rgba(0, 0, 0, 0.87);">
        <p>
          Welcome to our fishing club! We are going to hold our first ever fishing activity. Here you will find useful information in a nicely organized way.
        </p>
        <p>
          <!-- react-text: 14 -->
          We try to include as much detailed information as we can. Do let us know if we miss any information that you need. If you are in a hurry, please check
          <!-- /react-text -->
          <a href="/others/preparations"
             title="Preparations"
          >
            Preparations
          </a>
          <!-- react-text: 16 -->
          page where you will learn how to better prepare yourself for the activity.
          <!-- /react-text -->
        </p>
        <p>
          Hope you will enjoy the activity.
        </p>
        <br>
        <p>
          <small>
            <!-- react-text: 21 -->
            The source code of this website is hosted on
            <!-- /react-text -->
            <a href="https://github.com/ayltai/fishing-club"
               title="GitHub"
            >
              GitHub
            </a>
            <!-- react-text: 23 -->
            .
            <!-- /react-text -->
          </small>
        </p>
      </div>
    </div>
  </div>
</div>

`;
